Source/MaestroMakeEdgeScalars.cpp

```diff
@@ -0,0 +1,1100 @@
+
+#include <Maestro.H>
+#include <MaestroHydro_F.H>
+
+using namespace amrex;
+
+#if (AMREX_SPACEDIM == 2)
+
+#else
+
+void Maestro::MakeDivU(const Box& bx, 
+                       Array4<Real> const divu,
+                       Array4<Real> const umac,
+                       Array4<Real> const vmac,
+                       Array4<Real> const wmac,
+                       const Real* dx) 
+{
+    // timer for profiling
+    BL_PROFILE_VAR("Maestro::MakeDivU()",MakeDivU);
+
+    Real dx_local = dx[0];
+
+    AMREX_PARALLEL_FOR_3D(bx, i, j, k, 
+    {
+        divu(i,j,k) = 
+            umac(i+1,j,k) - umac(i,j,k) +
+            vmac(i,j+1,k) - vmac(i,j,k) +
+            wmac(i,j,k+1) - wmac(i,j,k);
+        divu(i,j,k) /= dx_local;
+    });
```

Comment:
not sure how it optimizes, but maybe precompute 1/dx_local outside the lambda and multiply